File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.String/String.ToMemoryStream.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.IO
Imports System.Text

Public Module Extensions_535

	''' <summary>
	'''     A string extension method that converts the @this to a MemoryStream.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>@this as a MemoryStream.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function ToMemoryStream(this As String) As Stream
		Dim encoding As Encoding = Activator.CreateInstance(Of ASCIIEncoding)()
		Return New MemoryStream(encoding.GetBytes(this))
	End Function
End Module

File: src/Z.Core/System.String/String.ToMemoryStream.cs
// Description: C# Extension Methods Library to enhances the .NET Framework by adding hundreds of new methods. It drastically increases developers productivity and code readability. Support C# and VB.NET
// Website & Documentation: https://github.com/zzzprojects/Z.ExtensionMethods
// Forum: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License: https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: http://www.zzzprojects.com/
// Copyright © ZZZ Projects Inc. 2014 - 2016. All rights reserved.
using System;
using System.IO;
using System.Text;

public static partial class Extensions
{
    /// <summary>
    ///     A string extension method that converts the @this to a MemoryStream.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>@this as a MemoryStream.</returns>
    public static Stream ToMemoryStream(this string @this)
    {
        Encoding encoding = Activator.CreateInstance<ASCIIEncoding>();
        return new MemoryStream(encoding.GetBytes(@this));
    }
}